examples/session/enumerate-software.py

```diff
@@ -0,0 +1,78 @@
+r"""List out NI software installed on the server machine.
```

Comment:
It'd be great to add this to the list of examples that are covered with validate_examples on the workflow here: https://github.com/ni/grpc-device/blob/d5d281adace5590683685c9cadea9d78046b3446/.github/workflows/validate_python.yml#L34

Note you can also use that script locally to check types and run the example (OR ... see why it's failing to install 😢 )